include/swift/SIL/SILInstruction.h

```diff
@@ -87,6 +88,83 @@ SILInstructionKind getSILInstructionKind(StringRef InstName);
 /// Map SILInstructionKind to a corresponding SILInstruction name.
 StringRef getSILInstructionName(SILInstructionKind Kind);
 
+/// A formal SIL reference to a list of values, suitable for use as the result
+/// of a SILInstruction.
+///
+/// *NOTE* Most multiple value instructions will not have many results, so if we
+/// want we can cache up to 3 bytes in the lower bits of the value.
+///
+/// *NOTE* Most of this defined out of line further down in the file to work
+/// around forward declaration issues.
+class SILInstructionResultArray {
+  const uint8_t *Pointer;
+  unsigned Size;
+  unsigned Stride;
+
+public:
+  SILInstructionResultArray(const NonValueInstruction *NVI);
```

Comment:
Just give it a default constructor that creates a zero-length array, I think.

Reply:
I did this on purpose. I wanted to try and use this to communicate to the user that one doesn't just default construct these. One needs an actual Instruction. That being said, perhaps I was going overboard.